lib/raw.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Although git-blame tells you this was written by bstack,
+ * it was really written by github.com/kristelteng. All credit
+ * for good parts go to her and all blame for bad parts go to
+ * bstack.
+ */
+var path = require('path');
+var _ = require('lodash');
+var through = require('through2');
+var gutil = require('gulp-util');
+var PluginError = gutil.PluginError;
+var File = require('vinyl');
+var fs = require('fs');
+var pug = require('pug');
+var marked = require('marked');
+
+/**
+ * Runs for each tarball. It:
+ * 1. generates html from json for references and schemas
+ * 2. generates index pages from README.md's
+ * 3. sets the appropriate paths for files, with tier information etc
+ */
+module.exports = function() {
+  // collect info for this particular project
+  // as the data stream is parsed
+  var projects = {};
+
+  // We moved this to pug because the ejs we use client-side
+  // isn't actually the normal ejs and it is annoying so I figured if
+  // I was going to port this to something else, may as well use what
+  // we use elsewhere.
+  var refTemplate = pug.compileFile('layout/reference.pug');
+
+  return through.obj(function each(file, enc, cb) {
+    if (file.isStream()) {
+      this.emit('error', new PluginError(PLUGIN_NAME, 'Streams are not supported!'));
+      return cb();
+    }
+
+    if (path.basename(file.relative) === 'metadata.json') {
+      return cb();
+    }
+    if (path.basename(file.relative) === 'README.md') {
+      file.path = file.path.slice(0, -9)
+    }
+    var project = file.relative.split('/')[1];
+    var type = file.relative.split('/')[2];
+    var spl = file.path.split(project);
+    // To guard for the project name existing in document names
+    var p = _.concat(spl[0], _.join(spl.slice(1), project));
+    var metadata = projects[project];
```

Comment:
Some comments would be helpful on all this splitting.  I think it's here to recognize certain files and directory structures in the tarball?  Is there a doc somewhere describing that tarball format?